core/trino-spi/src/main/java/io/trino/spi/classloader/ThreadContextClassLoader.java

```diff
@@ -31,4 +35,46 @@ public void close()
     {
         Thread.currentThread().setContextClassLoader(originalThreadContextClassLoader);
     }
+
+    public static void withClassLoaderOf(Class<?> clazz, Runnable runnable)
```

Comment:
I don’t think these “of” methods are used frequently enough to justify the API weight. Doing it in the caller seems more clear and isn’t much longer.